bin/wasm-node/javascript/src/index.js

```diff
@@ -79,11 +106,40 @@ export async function start(config) {
       // `expected` was pushed by the `addChain` method.
       // Resolve the promise that `addChain` returned to the user.
       expected.resolve({
+        health: () => {
+          // are we already waiting for a health message reply?
+          if (chainHealthRequestCallbacks[chainId]) {
+            const reject = chainHealthRequestCallbacks[chainId].reject;
+            delete chainHealthRequestCallbacks[chainId];
+            return reject(new Error('Called health before pending health request replied'));
+          }
+
+          // craft a new system_health messsage
+          const id = uuidv4();
+          const request = `{"id":"${id}","jsonrpc":"2.0","method":"system_health","params":[]}`;
+          let hrResolve, hrReject;
+          const hrPromise = new Promise((resolve, reject) => {
+            hrResolve = resolve;
+            hrReject = reject;
+          });
+
+          // track the new message
+          chainHealthRequestCallbacks[chainId] = {
+            requestId: id,
+            resolve: hrResolve,
+            reject: hrReject
+          };
+
+          // send it
+          worker.postMessage({ ty: 'request', request, chainId });
+          return hrPromise;
+        },
         sendJsonRpc: (request) => {
           if (workerError)
             throw workerError;
           if (!chainsJsonRpcCallbacks[message.chainId])
             throw new SmoldotError('Chain isn\'t capable of serving JSON-RPC requests');
+
           worker.postMessage({ ty: 'request', request, chainId });
```

Comment:
I think we should have a completely different namespace for regular JSON-RPC requests vs health requests.

Something like that:

```suggestion
          const parsed = JSON.parse(request);
          const newId = 'user:' + JSON.stringify(parsed.id);
          request.id = newId;
          worker.postMessage({ ty: 'request', request, chainId });
```

Reply:
Needs a `try ... catch` around `JSON.parse` which ignores invalid requests.